subsys/suit/mci/src/suit_mci_nrf9280.c

```diff
@@ -0,0 +1,664 @@
+	case SUIT_MANIFEST_SEC_TOP:
+	case SUIT_MANIFEST_SEC_SDFW:
+		if (key_id >= MANIFEST_PUBKEY_NRF_TOP_GEN0 &&
+		    key_id <= MANIFEST_PUBKEY_NRF_TOP_GEN0 + MANIFEST_PUBKEY_GEN_RANGE) {
+			return SUIT_PLAT_SUCCESS;
+		}
+		break;
+
+	case SUIT_MANIFEST_SEC_SYSCTRL:
+		if (key_id >= MANIFEST_PUBKEY_SYSCTRL_GEN0 &&
+		    key_id <= MANIFEST_PUBKEY_SYSCTRL_GEN0 + MANIFEST_PUBKEY_GEN_RANGE) {
+			return SUIT_PLAT_SUCCESS;
+		}
+		break;
+
+	case SUIT_MANIFEST_APP_ROOT:
+		if (key_id >= MANIFEST_PUBKEY_OEM_ROOT_GEN0 &&
+		    key_id <= MANIFEST_PUBKEY_OEM_ROOT_GEN0 + MANIFEST_PUBKEY_GEN_RANGE) {
+			return SUIT_PLAT_SUCCESS;
+		}
+		break;
+
+	case SUIT_MANIFEST_APP_RECOVERY:
+	case SUIT_MANIFEST_APP_LOCAL_1:
+	case SUIT_MANIFEST_APP_LOCAL_2:
+	case SUIT_MANIFEST_APP_LOCAL_3:
+		if (key_id >= MANIFEST_PUBKEY_APPLICATION_GEN0 &&
+		    key_id <= MANIFEST_PUBKEY_APPLICATION_GEN0 + MANIFEST_PUBKEY_GEN_RANGE) {
+			return SUIT_PLAT_SUCCESS;
+		}
+		break;
+
+	case SUIT_MANIFEST_RAD_RECOVERY:
+	case SUIT_MANIFEST_RAD_LOCAL_1:
+	case SUIT_MANIFEST_RAD_LOCAL_2:
+		if (key_id >= MANIFEST_PUBKEY_RADIO_GEN0 &&
+		    key_id <= MANIFEST_PUBKEY_RADIO_GEN0 + MANIFEST_PUBKEY_GEN_RANGE) {
+			return SUIT_PLAT_SUCCESS;
+		}
+		break;
+
+	default:
+		break;
+	}
+
+	return MCI_ERR_WRONGKEYID;
+}
+
+mci_err_t suit_mci_processor_start_rights_validate(const suit_manifest_class_id_t *class_id,
+						   int processor_id)
+{
+	if (class_id == NULL) {
+		return SUIT_PLAT_ERR_INVAL;
+	}
+
+	suit_manifest_role_t role = SUIT_MANIFEST_UNKNOWN;
+
+	if (suit_storage_mpi_role_get(class_id, &role) != SUIT_PLAT_SUCCESS) {
+		return MCI_ERR_MANIFESTCLASSID;
+	}
+
+	switch (role) {
+	case SUIT_MANIFEST_UNKNOWN:
+		return MCI_ERR_MANIFESTCLASSID;
+
+	case SUIT_MANIFEST_SEC_TOP:
+	case SUIT_MANIFEST_APP_ROOT:
+	case SUIT_MANIFEST_SEC_SDFW:
+		break;
+
+	case SUIT_MANIFEST_SEC_SYSCTRL:
+		/* Sys manifest */
+		if (processor_id == NRF_PROCESSOR_SYSCTRL) {
+			/* SysCtrl */
+			return SUIT_PLAT_SUCCESS;
+		}
+		break;
+
+	case SUIT_MANIFEST_APP_RECOVERY:
+	case SUIT_MANIFEST_APP_LOCAL_1:
+	case SUIT_MANIFEST_APP_LOCAL_2:
+	case SUIT_MANIFEST_APP_LOCAL_3:
+		/* App manifest */
+		if (processor_id == NRF_PROCESSOR_APPLICATION) {
+			/* Appcore */
+			return SUIT_PLAT_SUCCESS;
+		}
+		break;
+
+	case SUIT_MANIFEST_RAD_RECOVERY:
+	case SUIT_MANIFEST_RAD_LOCAL_1:
+	case SUIT_MANIFEST_RAD_LOCAL_2:
+		/* Rad manifest */
+		if (processor_id == NRF_PROCESSOR_RADIOCORE) {
+			/* Radiocore */
+			return SUIT_PLAT_SUCCESS;
+		}
+		break;
+
+	default:
+		break;
+	}
+
+	return MCI_ERR_NOACCESS;
+}
+
+mci_err_t suit_mci_memory_access_rights_validate(const suit_manifest_class_id_t *class_id,
+						 void *address, size_t mem_size)
+{
+	if (class_id == NULL || address == NULL || mem_size == 0) {
+		return SUIT_PLAT_ERR_INVAL;
+	}
+
+	suit_manifest_role_t role = SUIT_MANIFEST_UNKNOWN;
+
+	if (suit_storage_mpi_role_get(class_id, &role) != SUIT_PLAT_SUCCESS) {
+		return MCI_ERR_MANIFESTCLASSID;
+	}
```

Comment:
Please re-sync this file with nRF54H20 implementation.

Reply:
Fixed.